src/main/java/mantenimiento/codecounter/validators/logicalValidators/ControlStructureValidator.java

```diff
@@ -0,0 +1,43 @@
+package mantenimiento.codecounter.validators.logicalValidators;
+
+import mantenimiento.codecounter.interfaces.LogicalValidatorHandler;
+import mantenimiento.codecounter.constants.JavaRegextConstants;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * 
+ * Verifica la presencia de estructuras de control como if, for, while, switch, etc.
+ * Implementa la interfaz LogicalValidatorHandler para permitir encadenamiento de validadores.
+ */
+public class ControlStructureValidator implements LogicalValidatorHandler {
+    private LogicalValidatorHandler nextValidator;
```

Comment:
Aquí debes de extender de la clase `LogicalValidator` ya que esa es la que implementa la interfaz y así mismo, no requieres crear un atributo privado para almacenar la referencia al siguiente validador
```suggestion
public class ControlStructureValidator extends LogicalValidator {
```